### ISSUE TITLE
GitHub Discovery filter out forks

### ISSUE THREAD

**Issue by axdotl** [backstage/backstage#5178]
## Feature Suggestion

Enhance the GitHub discovery to filter out forks.

## Possible Implementation

- GitHub API offers a `type` filter, see [List organization repositories](https://docs.github.com/en/rest/reference/repos#list-organization-repositories) - maybe the same values for param `type` can be used for this feature
- Extend configuration options for locations
  ```yaml
  catalog:
    locations:
      - type: github-discovery
        target: https://github.com/myorg/*/blob/main/catalog-info.yaml
        options:
          - type: [all, public, private, forks, sources, member, internal]
          - ignoreForks: true
  ``` 
- Adjust `GithubDiscoveryProcessor` to consider configuration options

## Context

The discovery feature is really useful and we use it to discover repos for the whole org, but we also have some forks which we do not want to track in our Backstage App. Therefor it would be great to filter out forked repos.

**Comment by benjdlambert** [comment]
We're currently doing some work towards moving locations to entities so maybe that might make this part easier. Let's pick this conversation back up in a few weeks. +1

**Comment by freben** [comment]
We did the large engine update that he mentions some time ago, but I think that the ask in this issue is not addressed as such.

Locations in the config can't have additional data on them, so that particular encoding of the problem won't be supported either way. Hmm, this all sounds to me like it would make sense to rewrite the _processor_, to instead be an _entity provider_ which is a new mechanism in the rewritten catalog, for sourcing original entity data into the system. But we're still early in that journey so it'll take a little development to make it happen and it's not on the immediate near-term road map. Let me know if you feel you're up to dipping your toes into the development of this, and maybe we can offer some guidance.

**Comment by costasd** [comment]
Hi, we've hit that issue in my org couple days ago and thought to implement an intermediate solution to it in #15953 - until all the aforementioned bits and pieces for the more complete solution are in place. Hope it looks good, first time committing to backstage!

**Comment by benjdlambert** [comment]
Closing as resolved in #15953
